### PR TITLE
PERF: Improve iterfeatures performance

### PR DESCRIPTION
**Pull request by YuichiNotoya**
Fix for #863 .
I referred to @jorisvandenbossche 's code very much.
I implemented three different na strategies (null, drop, keep).
Under my PC environment, the speed improvement was nearly nine times as follows.

%timeit list(df.iterfeatures())
Before: 3.36 s ± 225 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
After : 375 ms ± 14.9 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

(The same condition as @jorisvandenbossche 's [Notebook](http://nbviewer.jupyter.org/gist/jorisvandenbossche/3bea109e0622fa5e3f90eecdc496c816) )

### CODE REVIEW

File: geopandas/tests/test_geodataframe.py [hunk 1]
```diff
@@ -607,6 +607,20 @@ def test_geodataframe_geointerface(self):
...
+        # when containing missing values
+        # null: ouput the missing entries as JSON null
+        result = list(df.iterfeatures(na='null'))[0]['properties']
+        assert result['BoroName'] is None
+        # drop: remove the property from the feature.
+        result = list(df.iterfeatures(na='drop'))[0]['properties']
+        assert 'BoroName' not in result.keys()
+        # keep: output the missing entries as NaN
+        result = list(df.iterfeatures(na='keep'))[0]['properties']
+        assert np.isnan(result['BoroName'])
```
Comment (jorisvandenbossche) [inline geopandas/tests/test_geodataframe.py]: Can you add one more specific test for checking that the other (non-null) features are python scalars and not numpy scalars? 

Like 

```
n [56]: df = geopandas.read_file(geopandas.datasets.get_path('nybb'))                                                                                                          

In [58]: type(df.loc[0, 'Shape_Leng'])                                                                                                                                          
Out[58]: numpy.float64

In [59]: type(df.loc[0, 'Shape_Leng']) is float                                                                                                                                 
Out[59]: False
```

The above should be true when accessing one of the elements in the features dict. 

And can you do this also when only having numerical columns? eg with `df[['Shape_Leng', 'Shape_Area', 'geometry']].iterfeatures()`

Comment (YuichiNotoya) [reply geopandas/tests/test_geodataframe.py]: Thank you for the review.
I pushed the changes following your comments.

Comment (jorisvandenbossche) [reply geopandas/tests/test_geodataframe.py]: Thanks for the update!

**Comment by jorisvandenbossche** [comment]
@YuichiNotoya Thanks a lot for the PR!

**Pull request merged**